libretroshare/src/serialiser/rstypeserializer.h

```diff
@@ -643,28 +732,32 @@ struct RsTypeSerializer
 		}
 		case RsGenericSerializer::FROM_JSON:
 		{
-			rapidjson::Document& jDoc(ctx.mJson);
+			RsJson& jDoc(ctx.mJson);
 			const char* mName = memberName.c_str();
-			ctx.mOk = ctx.mOk && jDoc.HasMember(mName);
+			bool hasMember = jDoc.HasMember(mName);
+			bool yielding = ctx.mFlags &
+			        RsGenericSerializer::SERIALIZATION_FLAG_YIELDING;
 
-			if(!ctx.mOk)
+			if(!hasMember)
 			{
-				std::cerr << __PRETTY_FUNCTION__ << " \"" << memberName
-				          << "\" not found in JSON:" << std::endl
-				          << jDoc << std::endl << std::endl;
-				print_stacktrace();
+				if(!yielding)
+				{
+					std::cerr << __PRETTY_FUNCTION__ << " \"" << memberName
+					          << "\" not found in JSON:" << std::endl
+					          << jDoc << std::endl << std::endl;
+					print_stacktrace();
+				}
+				ctx.mOk = false;
 				break;
 			}
 
 			rapidjson::Value& v = jDoc[mName];
 
-			RsGenericSerializer::SerializeContext lCtx(
-			            nullptr, 0, RsGenericSerializer::FORMAT_BINARY,
-			            RsGenericSerializer::SERIALIZATION_FLAG_NONE );
+			RsGenericSerializer::SerializeContext lCtx(nullptr, 0, ctx.mFlags);
 			lCtx.mJson.SetObject() = v; // Beware of move semantic!!
 
 			member.serial_process(j, lCtx);
-			ctx.mOk = ctx.mOk && lCtx.mOk;
+			ctx.mOk &= lCtx.mOk;
```

Comment:
Strictly speaking this is dangerous, unless you make sure by design that ctx.mOk is always 0 or 1, which might depend on the definition of true and false and on the particular implementation and on the value you put into ctx.mOk (some function return integers).  !4 is false, !2 is false as well, but !(4 & 2) is true. Normally you'd need &&= but I'm not sure this exists.

Reply:
Because ctx.mOk is a member of type bool in C++ (not in C) it is guaranted to be either true or false, so is is safe in that case, but i agree that the && variant is usually safer, and sadly &&= doesn't exists (we would benefit of that in other places especially because && is short circuit while & is not)